curl/src/client.cc

```diff
@@ -261,6 +261,33 @@ namespace leatherman { namespace curl {
         if (result != CURLE_OK) {
             throw http_request_exception(ctx.req, curl_easy_strerror(result));
         }
+        result = curl_easy_setopt(_handle, CURLOPT_SEEKFUNCTION, seek_body);
+        if (result != CURLE_OK) {
+            throw http_request_exception(ctx.req, curl_easy_strerror(result));
+        }
+        result = curl_easy_setopt(_handle, CURLOPT_SEEKDATA, &ctx);
+        if (result != CURLE_OK) {
+            throw http_request_exception(ctx.req, curl_easy_strerror(result));
+        }
+
+        switch (method) {
```

Comment:
I can see the argument both ways, but keeping setting the size of the data with setting the data seemed more coherent to me.